server/lib/authenticationUrl.js

```diff
@@ -23,8 +23,11 @@ module.exports = (app) => {
   }
 
   if (app.connection) {
-    loginUrl += (loginUrl.indexOf('?') >0) ? ((authProtocol === 'wsfed') ? '&whr=' : '&connection=') : ((authProtocol === 'wsfed') ? '?whr=' : '?connection=')
-    loginUrl += app.connection;
+    if (loginUrl.indexOf('?') > 0) {
+      loginUrl += (authProtocol === 'wsfed') ? '&whr=' : '&connection=';
+    } else {
+      loginUrl += (authProtocol === 'wsfed') ? '?whr=' : '?connection=';
+    }
```

Comment:
this is missing the actual connection - `loginUrl += app.connection;`

also, I think this logic would be must easier to read/understand if we broke it apart a bit, something like this:

```
const sep = loginUrl.indexOf('?') > 0 ? '?' : '&';
const type = authProtocol === 'wsfed' ? 'whr=' : 'connection=';
loginUrl += sep + type + app.connection;
```